app/src/main/java/net/squanchy/tweets/util/TweetTextFormatter.java

```diff
@@ -0,0 +1,59 @@
+package net.squanchy.tweets.util;
+
+import android.text.Spannable;
+import android.text.SpannableString;
+import android.text.style.URLSpan;
+
+import java.util.List;
+
+import net.squanchy.tweets.domain.view.HashtagEntity;
+import net.squanchy.tweets.domain.view.MentionEntity;
+import net.squanchy.tweets.domain.view.Tweet;
+import net.squanchy.tweets.domain.view.UrlEntity;
+import net.squanchy.tweets.model.ParsedTweet;
+import net.squanchy.tweets.parsing.TweetTextParser;
+
+public class TweetTextFormatter {
+
+    private static final String BASE_TWITTER_URL = "https://twitter.com/";
+    private static final String MENTION_URL_TEMPLATE = BASE_TWITTER_URL + "%s";
+    private static final String QUERY_URL_TEMPLATE = BASE_TWITTER_URL + "search?q=%s";
+
+    private final TweetTextParser tweetTextParser;
+
+    public TweetTextFormatter(TweetTextParser tweetTextParser) {
+        this.tweetTextParser = tweetTextParser;
+    }
+
+    public Spannable format(Tweet tweet) {
+        Spannable spannable = new SpannableString(tweet.text());
+
+        ParsedTweet parsedTweet = tweetTextParser.parse(tweet.text());
+
+        addUrlSpans(spannable, parsedTweet.urls());
```

Comment:
Why do we use 3 different classes (mention, hashtag and url) if then we leak the info (the template for the url) here? :tongue: